### PR TITLE
LWM2M: Fix check_authorization() for Access Control Objects

### PR DESCRIPTION
**Pull request by bsmelo**
I'm improving the `src/test/test-lwm2m.c` program to cover all new functionalities
added during GSoC, and found this issue.

This patch fixes a bug when a given LWM2M Server who owns a given
Access Control Object Instance tries to modify it (for instance,
to add or remove an ACL for a given third-party LWM2M Server).

There was no way to modify an Access Control Object Instance, since
the check_authorization() would fail, not finding an Access Control
Object Instance related to the target instance - which should not
exist anyway. To fix it, a new condition was added in case the target
`obj_id` == `ACCESS_CONTROL_OBJECT_ID`, in which we should only look
for the target `instance_id` itself and then inside the OWNER Resource,
not looking for ACLs or anything else.

Signed-off-by: Bruno Melo <bsilva.melo@gmail.com>

### CODE REVIEW

**Comment by cabelitos** [comment]
lgtm

File: src/lib/comms/sol-lwm2m-client.c [hunk 1]
```diff
@@ -452,14 +452,36 @@ check_authorization(struct sol_lwm2m_client *client,
...
+                }
+
+                if (res[0].data[0].content.integer == server_id)
+                    r = 1;
+                else
+                    r = 0;
+                goto exit_clear_1;
+            }
+        }
+    }
```
Comment (cabelitos) [inline src/lib/comms/sol-lwm2m-client.c]: I guess if the instance was not found we can return an error as well.

Comment (bsmelo) [inline src/lib/comms/sol-lwm2m-client.c]: Fixed this (and added `return -EINVAL` at the end of the "normal" case as well).

Should I make a new PR or update this one?

Comment (cabelitos) [inline src/lib/comms/sol-lwm2m-client.c]: Send a new one

**Comment by bsmelo** [comment]
An updated version is at #2282. Thanks!

**Pull request closed without merging**